File: javascript/easy/TwoSum.js
/**
 * Given nums = [2, 7, 11, 15], target = 9, */
var arr = [1, 2, 19, 28];
var target = 21;
var twoSum = function (nums, target) {
  for (var i = 0; i < nums.length; i++) {
    for (var j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] == target) {
        return [i, j];
      }
    }
  }
};
console.log(twoSum(arr, target));

File: python/easy/1week/1day/TwoSum.py
''':arg
Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

Example:

Given nums = [2, 7, 11, 15], target = 9,

Because nums[0] + nums[1] = 2 + 7 = 9,
return [0, 1].
'''


class Solution(object):
    def twoSum(self, nums, target):
        for n1 in range(0, len(nums)):
            for n2 in range(1, len(nums)):
                if nums[n1] + nums[n2] == target and n1 != n2:
                    return [n1, n2]


class Solution(object):
    def twoSum(self, nums, target):
        lookup = {}
        for index, item in enumerate(nums):
            if (target - item) in lookup:
                return [lookup[target - item], index]
            else:
                lookup[item] = index


